src/e2e-test/java/io.cdap.plugin/utils/ValidationHelper.java

```diff
@@ -0,0 +1,149 @@
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class ValidationHelper {
+
+    public static List<Map<String, Object>> getBigQueryRecordsAsMap(String projectId, String database, String tableName)
+            throws IOException, InterruptedException {
+        String query = "SELECT * EXCEPT ( _row_id, _source_timestamp, _sort) FROM `" + projectId
+                + "." + database + "." + tableName + "`";
+        List<Map<String, Object>> bqRecords = new ArrayList<>();
+        //Let us account for 90 seconds flush here.And for every flush, how long does it take to reflect the changes in bg by running queries.
+        //handle exceptions and retry
+        RetryPolicy<Object> retryPolicy = new RetryPolicy<>()
+                .withMaxAttempts(5)
+                .onRetry(  e -> System.out.println("Retrying exception"))
+                .withDelay(Duration.ofSeconds(60));
+                //.withMaxDuration(Duration.of(5, ChronoUnit.MINUTES));
+        TableResult results = Failsafe.with(retryPolicy).get(() -> {
+            //Let changes reflect in bq making sure flush cycle is included
+//            TimeUnit time = TimeUnit.SECONDS;
+//            time.sleep(180);
+            return BigQueryClient.getQueryResult(query);
+        });
+        List<String> columns = new ArrayList<>();
+        for (Field field : results.getSchema().getFields()) {
+            columns.add(field.getName() + "#" + field.getType());
+        }
+        for (FieldValueList row : results.getValues()) {
+            Map<String, Object> record = new HashMap<>();
+            int index = 0;
+            for (FieldValue fieldValue : row) {
+                String columnName = columns.get(index).split("#")[0];
+                String dataType = columns.get(index).split("#")[1];
+                Object value;
+                if(dataType.equalsIgnoreCase("TIMESTAMP")){
+                    value = fieldValue.getTimestampValue();
+//                    } else if (dataType.toUpperCase().equals("FLOAT")) {
+//                        value = Float.parseFloat(fieldValue.getStringValue());
+                } else {
+                    value = fieldValue.getValue();
+                    value = value != null ? value.toString() : null;
+                }
+                record.put(columnName, value);
+                index++;
+            }
+            bqRecords.add(record);
+        }
+        return bqRecords;
+    }
+
+    public static void validateRecords(List<Map<String, Object>> sourceOracleRecords,
+                                       List<Map<String, Object>> targetBigQueryRecords) {
+
+        String uniqueField = PluginPropertyUtils.pluginProp("primaryKey");
+        //Assert.assertEquals("Checking the size of lists ", targetBigQueryRecords.size(), sourceOracleRecords.size()); //doesn't make sense in our case, since few datatypes are dropped during assessment
```

Comment:
How does dropping of data types impact list size here which represent number of rows?